pkg/controller/datavolume/smart-clone-controller.go

```diff
@@ -347,7 +352,8 @@ func (r *SmartCloneReconciler) getTargetPVC(dataVolume *cdiv1.DataVolume) (*core
 	return pvc, nil
 }
 
-func newPvcFromSnapshot(snapshot *snapshotv1.VolumeSnapshot, targetPvcSpec *corev1.PersistentVolumeClaimSpec) (*corev1.PersistentVolumeClaim, error) {
+func newPvcFromSnapshot(name string, snapshot *snapshotv1.VolumeSnapshot, targetPvcSpec *corev1.PersistentVolumeClaimSpec) (*corev1.PersistentVolumeClaim, error) {
+	targetPvcSpecCopy := targetPvcSpec.DeepCopy()
```

Comment:
why do we need `targetPvcSpecCopy`?

Reply:
we used to mutate the pvcSpec through the ptr (accidentally)
https://github.com/kubevirt/containerized-data-importer/blob/bfe30a89648bdf37e8fd2eafcb65a04e9885cc20/pkg/controller/datavolume/smart-clone-controller.go#L392-L400